### DISCUSSION TITLE
Is there a way to include a user specified date with `googleapiclient` when doing a Google search?

### DISCUSSION THREAD

**Discussion by leockl** [googleapis/google-api-python-client#2179] [Q&A]
Anyone know if there is a way to enable a user to specify/filter a date to only search for posts which are posted on a particular date when using `googleapiclient`? 

I am thinking this may be possible because when we go onto Google's page and search for something, on the next page where it's displaying all the searched links, on the top right there is a "Tools" button that you can click on -> click "Any Time" -> click "Custom range" and this will let you specify the date of all the links that was posted on a particular date.

I was hoping that there would be a `date` parameter in `googleapiclient` but when I ran my code below which has the `date` parameter, it comes up with an error `TypeError: Got an unexpected keyword argument date`. The code works when I remove the `date` parameter.

```
from googleapiclient.discovery import build

class GoogleChat():
    def __init__(self):
        # init google search engine
        self.service = build(
            # https://developers.google.com/custom-search/v1/introduction
            "customsearch", "v1", developerKey="MY-DEVELOPER-KEY"
        )

    def _search(self, query, date):
        # call google search api
        response = (
            self.service.cse()
            .list(
                q=query,
                cx="MY-SEARCH-ENGINE-ID", # https://programmablesearchengine.google.com/controlpanel/all
                date=date
            )
            .execute()
        )
        return response['items']
```

Would greatly appreciate it if anyone could help.

**Comment by Cheril311** [comment]
Did you get any solution?
